### PR TITLE
feat(psl): accept `@prisma/mysql` provider only when the `nodeDrivers` feature flag is enabled

### PR DESCRIPTION
**Pull request by jkomyno**
Closes https://github.com/prisma/team-orm/issues/55

### CODE REVIEW

File: psl/builtin-connectors/src/mysql_datamodel_connector.rs [hunk 1]
```diff
@@ -237,6 +237,20 @@ impl Connector for MySqlDatamodelConnector {
...
+    fn validate_datasource(
+        &self,
+        preview_features: BitFlags<PreviewFeature>,
+        ds: &Datasource,
+        errors: &mut Diagnostics,
+    ) {
+        if ds.provider == "@prisma/mysql" && !preview_features.contains(PreviewFeature::NodeDrivers) {
+            errors.push_error(DatamodelError::new_validation_error(
+                "The provider \"@prisma/mysql\" is currently in preview and must be explicitly enabled. Please add the preview feature flag \"nodeDrivers\" to the generator block in your schema.prisma file.",
+                Span::empty(),
```
Comment (aqrln) [inline psl/builtin-connectors/src/mysql_datamodel_connector.rs:249]: should we maybe add `provider_span` to `Datasource` and use it here?

Comment (jkomyno) [reply psl/builtin-connectors/src/mysql_datamodel_connector.rs:249]: Hi @aqrln, how so?

**Comment by janpio** [comment]
https://github.com/prisma/team-orm/issues/55#issuecomment-1564673944

**Review by miguelff — APPROVED** [review]
I'm approving this after having addressed @aqrln's feedback. Nevertheless I only did this because it was a way for me to sneak peak into this part of the codebase and learn. This patch will gain priority only after we decide that our current approach to nodeJS drivers make sense.

**Review by miguelff — APPROVED** [review]
I'm approving this after having addressed @aqrln's feedback. Nevertheless I only did this because it was a way for me to sneak peak into this part of the codebase and learn. This patch will gain priority only after we decide that our current approach to nodeJS drivers make sense.

**Comment by jkomyno** [comment]
Closed as deprecated.

**Pull request closed without merging**